extensions/wikia/Rail/RailController.class.php

```diff
@@ -8,139 +8,159 @@ class RailController extends WikiaController {
+		$this->isEditPage = isset( $params['isEditPage'] ) ? $params['isEditPage'] : false;
+		$this->railModuleList = $this->filterModules( $railModules, self::FILTER_NON_LAZY_MODULES );
 		$this->isGridLayoutEnabled = BodyController::isGridLayoutEnabled();
 		$this->isAside = $this->wg->RailInAside;
 		$this->loadLazyRail = $railModules > $this->railModuleList;
-
-		wfProfileOut(__METHOD__);
 	}
 
 	/**
 	 * Entry point for lazy loading right rail for anon users
 	 */
 	public function executeLazyForAnons() {
-		wfProfileIn(__METHOD__);
-
 		$this->getLazyRail();
 
-		$this->response->setCacheValidity(WikiaResponse::CACHE_STANDARD);
-
-		wfProfileOut(__METHOD__);
+		$this->response->setCacheValidity( WikiaResponse::CACHE_STANDARD );
 	}
 
 	/**
 	 * Entry point for lazy loading right rail for logged in users
 	 */
 	public function executeLazy() {
-		wfProfileIn(__METHOD__);
-
 		$this->getLazyRail();
-
-		//$this->response->setCacheValidity(null, null, []); // NOP
-
-		wfProfileOut(__METHOD__);
 	}
 
 	/**
 	 * Get lazy right rail modules
 	 */
 	protected function getLazyRail() {
-		wfProfileIn(__METHOD__);
-		global $wgUseSiteJs, $wgAllowUserJs, $wgTitle, $wgAllInOne;
-		$title = Title::newFromText($this->request->getVal('articleTitle', null), $this->request->getInt('namespace', null));
-
-		if ($title instanceof Title) {
-			// override original wgTitle from title given in parameters
-			// we cannot use wgTitle that is created on by API because it's broken on wikis without '/wiki' in URL
-			// https://wikia-inc.atlassian.net/browse/BAC-906
-			$oldWgTitle = $wgTitle;
-			$wgTitle = $title;
-			$assetManager = AssetsManager::getInstance();
-			$railModules = $this->filterModules((new BodyController)->getRailModuleList(), self::FILTER_LAZY_MODULES);
-			$this->railLazyContent = '';
-			krsort($railModules);
-			foreach ($railModules as $railModule) {
-				$this->railLazyContent .= $this->app->renderView(
-					$railModule[0], /* Controller */
-					$railModule[1], /* Method */
-					$railModule[2] /* array of params */
-				);
-			}
+		global $wgUseSiteJs, $wgAllowUserJs, $wgTitle, $wgAllInOne, $wgUser;
+		$title = Title::newFromText(
+			$this->request->getVal( 'articleTitle', null ),
+			$this->request->getInt( 'namespace', null )
+		);
+
+		if ( !( $title instanceof Title ) ) {
+			return;
+		}
 
-			$this->railLazyContent .= Html::element('div', ['id' => 'WikiaAdInContentPlaceHolder']);
+		// override original wgTitle from title given in parameters
+		// we cannot use wgTitle that is created on by API because it's broken on wikis without '/wiki' in URL
+		// https://wikia-inc.atlassian.net/browse/BAC-906
+		$oldWgTitle = $wgTitle;
+		$wgTitle = $title;
+		$assetManager = AssetsManager::getInstance();
+		$railModules = $this->filterModules(
+			( new BodyController )->getRailModuleList(),
+			self::FILTER_LAZY_MODULES
+		);
+		$this->railLazyContent = '';
+
+		krsort( $railModules );
+
+		// TODO XW-2760 remove after experiment is done
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this "TODO" comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)